centos-7/ks.cfg

```diff
@@ -0,0 +1,268 @@
+cat << PRODUCT > /etc/product
+Name: Joyent Instance
+Image: $NAME $BUILDDATE
+Documentation: $DOCS
+Description: $NAME 64-bit image with just essential packages installed. Ideal for users who are comfortable with setting up their own environment and tools.
+PRODUCT
+
+# Grub
+echo "Installing new GRUB file"
+cat <<EOF > /etc/default/grub
+GRUB_TIMEOUT=5
+GRUB_DISTRIBUTOR="CentOS Linux"
+GRUB_DEFAULT=saved
+GRUB_DISABLE_SUBMENU=true
+GRUB_TERMINAL="serial console"
+GRUB_SERIAL_COMMAND="serial --speed=115200 --unit=0 --word=8 --parity=no --stop=1"
+GRUB_CMDLINE_LINUX="tsc=reliable divider=10 plymouth.enable=0 console=ttyS0 crashkernel=auto"
+GRUB_DISABLE_RECOVERY="true"
+EOF
+/usr/sbin/grub2-mkconfig -o /boot/grub2/grub.cfg
+
+# Make sure we have the latest security updates
+echo "Updating packages"
+/usr/bin/yum clean all
+/usr/bin/yum update -y
+
+# Install Node.js and json via EPEL
+/usr/bin/yum install -y epel-release
+/usr/bin/yum install -y nodejs
+/usr/bin/npm install -g json
+
+packages="/run/install/joyks/Packages"
+for file in $packages/*.rpm; do
+    [[ -f $file ]] && yum install -y "$file"
+done
+
+echo "Configuring cloud-init"
+cat <<EOF > /etc/cloud/cloud.cfg.d/90_smartos.cfg
+datasource_list: [ SmartOS ]
+
+# Preserve traditional root@<ip> login that was possible with rc.local
+disable_root: false
+
+# Do not create the centos/ubuntu/debian user
+users: [ ]
+
+mounts:
+- [ vdb, /data, auto, "defaults,nofail" ]
+EOF
+
+# Clean up all yum caches
+echo "Cleaning up yum caches"
+/usr/bin/yum clean all
+
+# Clean up network devices
+echo "Cleaning up network devices"
+/bin/rm -f /etc/udev/rules.d/70-persistent-net.rules
+/bin/rm -rf /etc/sysconfig/network-scripts/ifcfg-*
+/bin/find /var/lib/dhclient -type f -exec rm -f '{}' +
+/bin/rm -f /etc/resolv.conf
+
+# Remove hostname
+echo "Clearing out /etc/hostname"
+cat /dev/null > /etc/hostname
+
+# Tune Linux vm.dirty_background_bytes (IMAGE-439)
+# The following tuning causes dirty data to begin to be background flushed at
+# 100 Mbytes, so that it writes earlier and more often to avoid a large build
+# up and improving overall throughput.
+echo "Setting vm.dirty_background_bytes"
+echo "vm.dirty_background_bytes=100000000" >> /etc/sysctl.conf
+
+# Disable password auth. SSH login is via ssh key only. A password is being set
+# for root via the image manifest per IMAGE-459.
+echo "Disabling password auth in sshd_config"
+sed s/PasswordAuthentication\ yes/PasswordAuthentication\ no/ -i /etc/ssh/sshd_config
+
+# Disable kdump
+echo "Disabling kdump"
+systemctl disable kdump.service
+
+# Ensure we have sane and consistent defaults for ntp.conf
+sed s/restrict\ default\ nomodify\ notrap\ nopeer\ noquery/restrict\ default\ kod\ nomodify\ notrap\ nopeer\ noquery/ -i /etc/ntp.conf
+# For IPv6
+echo "restrict -6 default kod nomodify notrap nopeer noquery" >> /etc/ntp.conf
+sed s/restrict\ ::1/restrict\ -6\ ::1/ -i /etc/ntp.conf
+
+# Clean up files
+echo "Ceaning up build files"
```

Comment:
Looks like a case of s/Ceaning/Cleaning/g.

Reply:
ok